meta-balena-common/recipes-containers/balena/balena/balena-host.service

```diff
@@ -13,8 +13,14 @@ EnvironmentFile=-/etc/docker/balenahost.env
 ExecStart=/usr/bin/balenad --delta-data-root=/mnt/sysroot/active/balena --delta-storage-driver=@BALENA_STORAGE@ --log-driver=journald -s @BALENA_STORAGE@ --data-root=/mnt/sysroot/inactive/balena -H fd:// --pidfile=/var/run/balena-host.pid --exec-root=/var/run/balena-host --bip 10.114.101.1/24 --fixed-cidr=10.114.101.128/25 --iptables=false --max-download-attempts=10 --exec-opt native.cgroupdriver=systemd
 #Adjust OOMscore to -900 to make killing unlikely
 OOMScoreAdjust=-900
-MountFlags=slave
 LimitNOFILE=1048576
-LimitNPROC=1048576
+LimitNPROC=infinity
 LimitCORE=infinity
+TasksMax=infinity
```

Comment:
Related to this change we have https://github.com/balena-os/meta-balena/issues/1844. See the comment about the transient systemd unit that balena-engine creates - this setting does not apply to it so the limitation is not lifted.

Reply:
I don't understand why it won't work for us, this is the recommended upstream solution https://success.docker.com/article/how-to-reserve-resource-temporarily-unavailable-errors-due-to-tasksmax-setting

I would also like your opinion on removing the mount flags option, see the links I added to the commit body for reasons why/upstream discussions

Reply:
Hi @robertgzr, this only sets the TaskMax for the balena-engine. Systemd units that are created by the engine using the d-bus API are not affected by this change. So the change is trying to lift the limit on the maximum number of threads that a container can spawn, but it doesn't. We also need to manually lift it for the transient units created by the engine with:
```
systemctl set-property docker-<container-id>.scope TasksMax=infinity
```
I think this fix should go in, I was just pointing out that it does not completely achieve what it is meant to do. I don't know if the transient units are also used by docker or are balena-engine specific.

Reply:
@robertgzr About the `MountFlags=slave` removal, I have read the relevant documentation several times and as nearly always with systemd it's not completely clear to me. It seems to imply that removing this option will not affect the behaviour, and it actually recommends to leave the default. So if the change comes from upstream I am probably ok as long as we specifically test volumes and other mounts that could be affect by the propagation change.

Reply:
This has now been separated to a different PR so we can review the changes there.